src/google/protobuf/testing/googletest.cc

```diff
@@ -94,6 +94,10 @@ string TestSourceDir() {
 namespace {
 
 string GetTemporaryDirectoryName() {
+  char *from_environment = getenv("TEST_TMPDIR");
```

Comment:
I think there are other projects that are also writing to /tmp in unit tests. When using bazel to build these projects, does bazel show some kind of error message to warn about this? When I use bazel to build protobuf, it builds and runs fine for me, so I don't know how users of bazel can find out this problem.